internal/services/newrelic/new_relic_monitor_resource.go

```diff
@@ -537,3 +633,86 @@ func flattenUserInfoModel(input *monitors.UserInfo) []UserInfoModel {
 
 	return append(outputList, output)
 }
+
+func expandMonitorSubscriptionList(input []NewRelicMonitoredSubscription) *[]monitoredsubscriptions.MonitoredSubscription {
+	results := make([]monitoredsubscriptions.MonitoredSubscription, 0)
+	if len(input) == 0 {
+		return &results
+	}
+
+	for _, v := range input {
+		results = append(results, monitoredsubscriptions.MonitoredSubscription{
+			SubscriptionId: pointer.To(v.SubscriptionId),
+		})
+	}
+
+	return &results
+}
+
+func flattenMonitorSubscriptionList(input *[]monitoredsubscriptions.MonitoredSubscription) []NewRelicMonitoredSubscription {
+	results := make([]NewRelicMonitoredSubscription, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, v := range *input {
+		results = append(results, NewRelicMonitoredSubscription{
+			// The returned subscription UUID is in upper case
+			SubscriptionId: strings.ToLower(pointer.From(v.SubscriptionId)),
+		})
+	}
+
+	return results
+}
+
+func resourceMonitoredSubscriptionsWaitForAvailable(ctx context.Context, client *monitoredsubscriptions.MonitoredSubscriptionsClient, id monitoredsubscriptions.MonitorId, monitoredSubscriptions []NewRelicMonitoredSubscription) error {
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		return fmt.Errorf("internal error: context had no deadline")
+	}
+	state := &pluginsdk.StateChangeConf{
+		MinTimeout:                5 * time.Second,
+		ContinuousTargetOccurence: 2,
+		Pending:                   []string{"Unavailable"},
+		Target:                    []string{"Available"},
+		Refresh:                   resourceMonitoredSubscriptionsRefresh(ctx, client, id, monitoredSubscriptions),
+		Timeout:                   time.Until(deadline),
+	}
+
+	if _, err := state.WaitForStateContext(ctx); err != nil {
+		return fmt.Errorf("waiting for NewRelic Monitored Subscriptions of %s to become available: %+v", id, err)
+	}
+
+	return nil
+}
+
+func resourceMonitoredSubscriptionsRefresh(ctx context.Context, client *monitoredsubscriptions.MonitoredSubscriptionsClient, id monitoredsubscriptions.MonitorId, monitoredSubscriptions []NewRelicMonitoredSubscription) pluginsdk.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		log.Printf("[DEBUG] Checking to see if NewRelic Monitored Subscriptions of %s is available ..", id)
+
+		resp, err := client.Get(ctx, id)
+		if err != nil {
+			return resp, "Error", fmt.Errorf("retrieving NewRelic Monitored Subscriptions of %s: %+v", id, err)
```

Comment:
The `id` is created by `monitoredsubscriptions` , not `monitors`, other logs also have the same problem

Reply:
The error message should be OK, because there is no ID struct for `monitoredsubscriptions`, the line 693 is retrieving the monitoredsubscriptions, I have renamed the client to `monitoredSubscriptionClient` to make it more clear.